### PR TITLE
Installed devise

### PR DESCRIPTION
**Pull request by emico7**
My current test coverage is: 98%

### CODE REVIEW

File: app/assets/javascripts/welcome.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (brittanyjspringer) [inline app/assets/javascripts/welcome.coffee]: @emico7 delete please!

File: app/assets/stylesheets/welcome.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the welcome controller here.
```
Comment (brittanyjspringer) [inline app/assets/stylesheets/welcome.scss]: Delete please

File: config/initializers/devise.rb [hunk 3]
```diff
@@ -0,0 +1,274 @@
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = 'eca6551514263b7a3017e9e4ab2ad2a14ac67a4662cbdcf9c04056ffe558901b5d822e7c5faeca52bf6a30171b10b42368cf1998cddb9af5379faf570bbf2874'
```
Comment (brittanyjspringer) [inline config/initializers/devise.rb]: Uncomment this please.

**Comment by emico7** [comment]
@wonderwoman13 Now users can sign in with username or email.

My current test coverage is: 98%

**Comment by brittanyjspringer** [comment]
Looks great! 

popkey

**Pull request merged**